python/dgl/distributed/partition.py

```diff
@@ -42,6 +44,49 @@ def _get_part_ranges(id_ranges):
         res[key] = np.concatenate([np.array(l) for l in id_ranges[key]]).reshape(-1, 2)
     return res
 
+
+def _convert_str_to_tuple_key(part_d):
+    copy_d = copy.deepcopy(part_d)
+    # Iterate over d, modify on copy_d
+
+    def _recursive_convert_str_to_tuple_key(part_d, key_tuple=tuple()):
+        if isinstance(part_d, dict):
+            for k, v in part_d.items():
+                sub_d = copy_d
+                for key in key_tuple:
+                    sub_d = sub_d[key]
+                if k.startswith("tuple"):
```

Comment:
What if someone names their edge type "tuple" (though unlikely)?

Is it possible to always encode the canonical edge types in JSON?

Reply:
NVM the edge type keys are always replaced with canonical edge types so this is not a problem.